### PR TITLE
implement shell function in Agent

### PR DESCRIPTION
**Pull request by AndyWu2015**

### CODE REVIEW

**Comment by AndyWu2015** [comment]
Hey @gamerson  , we can use this for executing "lb -s some-bsn" instead of getting all bundles.

**Comment by pkriens** [comment]
LGTM

File: biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java [hunk 1]
```diff
@@ -132,4 +135,17 @@ public PrintStream getOut() {
...
 
+	public String getLastOutput() {
+		return lastOutput;
+	}
+
+	public void setLastOutput(String out) {
+		if (!"".equals(out) && out != null) {
+			out = out.replaceAll("^>.*$", "");
+
+			if (!"".equals(out) && !out.startsWith("true")) {
```
Comment (gamerson) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: if the shell command returns 'true' , do we not want to just let "true" be returned?

also the "^>.*$" assumes that the gogo prompt will be standard, but what if its modified?  with

```
g! prompt= '$ '
$   
```

Comment (AndyWu2015) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: yes , but I think we can only assume that user will not change the prompt string when they calling Agent.shell("string") , also it makes no sense that user change the prompt string cause they just want to get gogo shell result and they are not using command line and don't need to see the prompt in UI.

Comment (gamerson) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: I think that we can go ahead and merge this and if we need improved implementation in the future, we can always improve it.

**Comment by gamerson** [comment]
hey @AndyWu2015  can you test this new commit I pushed to your branch?  if it works ok I will merge

**Comment by AndyWu2015** [comment]
Sure , I tested it on my side , it works well.

**Pull request merged**